firefox-ios/Ecosia/Account/Auth/Auth.swift

```diff
@@ -351,4 +361,65 @@ extension Auth {
         webView.configuration.websiteDataStore.httpCookieStore.setCookie(sessionTokenCookie)
         print("Session token cookie set successfully")
     }
+    
+    // MARK: - State Management Integration
+    
+    /**
+     Dispatches authentication state changes to the new state management system
+     and posts legacy notifications for backward compatibility.
+     
+     - Parameters:
+     -   isLoggedIn: Current authentication status
+     -   fromCredentialRetrieval: Whether this is from credential retrieval (for state loaded)
+     */
+    private func dispatchAuthStateChange(isLoggedIn: Bool, fromCredentialRetrieval: Bool) async {
+        // Determine the correct action type
+        let actionType: EcosiaAuthActionType
+        if fromCredentialRetrieval {
+            actionType = .authStateLoaded
+        } else if isLoggedIn {
+            actionType = .userLoggedIn
+        } else {
+            actionType = .userLoggedOut
+        }
+        
+        // Dispatch to the new state management system
+        AuthStateManager.shared.dispatchAuthState(isLoggedIn: isLoggedIn, actionType: actionType)
+        
+        // Post legacy notifications for backward compatibility
+        await postLegacyNotifications(isLoggedIn: isLoggedIn, actionType: actionType)
+    }
+    
+    /**
+     Posts legacy notifications for backward compatibility with existing code.
+     
+     - Parameters:
+     -   isLoggedIn: Current authentication status
+     -   actionType: The type of auth action that occurred
+     */
+    private func postLegacyNotifications(isLoggedIn: Bool, actionType: EcosiaAuthActionType) async {
```

Comment:
Not sure I get why these legacy notifications are needed. Weren't them just introduced  on the `AuthStateAction` file below? Are they being used anywhere?